trl/trainer/online_dpo_trainer.py

```diff
@@ -591,9 +591,23 @@ def __init__(
                 for i, reward_func in enumerate(self.reward_funcs):
                     if isinstance(reward_func, PreTrainedModel):
                         self.reward_funcs[i] = prepare_deepspeed(reward_func, self.accelerator)
+        elif self.is_fsdp_enabled:
+            # Prepare ref model for FSDP training
+            if self.ref_model is not None:
+                self.ref_model = prepare_fsdp(self.ref_model, self.accelerator)
+            if self.reward_funcs is not None:
+                for i, reward_func in enumerate(self.reward_funcs):
+                    if isinstance(reward_func, PreTrainedModel):
+                        # Set device placement to True to make `prepare_model` move `reward_func` to device when using fsdp
+                        self.reward_funcs[i] = self.accelerator.prepare_model(
+                            reward_func, evaluation_mode=True, device_placement=True
+                        )
```

Comment:
Refactoring, but feel free to ignore it:
```suggestion
                # Set device placement to True to make `prepare_model` move `reward_func` to device when using fsdp
                self.reward_funcs = [
                    self.accelerator.prepare_model(reward_func, evaluation_mode=True, device_placement=True)
                    if isinstance(reward_func, PreTrainedModel)
                    else reward_func
                    for reward_func in self.reward_funcs
                ]
```